rocAL/rocAL/source/pipeline/node.cpp

```diff
@@ -37,17 +38,26 @@ Node::create(std::shared_ptr<Graph> graph)
 
     _graph = graph;
 
-    if(!_inputs.empty())
+    if(!_inputs.empty() && !_outputs.empty())
     {
-        vx_status width_status, height_status;
-        std::vector<uint32_t> roi_width(_batch_size, _inputs[0]->info().width());
-        std::vector<uint32_t> roi_height(_batch_size, _inputs[0]->info().height_single());
-        _src_roi_width = vxCreateArray(vxGetContext((vx_reference) _graph->get()), VX_TYPE_UINT32, _batch_size);
-        _src_roi_height = vxCreateArray(vxGetContext((vx_reference) _graph->get()), VX_TYPE_UINT32, _batch_size);
-        width_status = vxAddArrayItems(_src_roi_width, _batch_size, roi_width.data(), sizeof(vx_uint32));
-        height_status = vxAddArrayItems(_src_roi_height, _batch_size, roi_height.data(), sizeof(vx_uint32));
-        if (width_status != 0 || height_status != 0)
-            THROW(" vxAddArrayItems failed : " + TOSTR(width_status) + "  " + TOSTR(height_status))
+        vx_size stride[2];
+        std::vector<size_t> roi_dims = {_batch_size, 4};
+        stride[0] = sizeof(vx_uint32);
+        stride[1] = stride[0] * roi_dims[0];
+        vx_enum mem_type = VX_MEMORY_TYPE_HOST;
+        if (_inputs[0]->info().mem_type() == RocalMemType::HIP)
+            mem_type = VX_MEMORY_TYPE_HIP;
+            
+        _src_tensor_roi = vxCreateTensorFromHandle(vxGetContext((vx_reference) _graph->get()), 2, roi_dims.data(), VX_TYPE_UINT32, 0, 
```

Comment:
Can we not remove the hardcoded value 2 here & add a generic queried value ?

Reply:
We cannot add a generic queried value, Since we need to specify the num of dims
Will add a varible called num_of_dims for readability

Reply:
Done